src/inmanta/protocol/methods.py

```diff
@@ -238,6 +238,30 @@ def create_token(tid: uuid.UUID, client_types: list, idempotent: bool = True):
     """
 
 
+#  Halt all orchestrator operations for an environment
+
+
+@typedmethod(path="/halt", operation="POST", arg_options=ENV_OPTS, client_types=[const.ClientType.api], api_version=1)
```

Comment:
I followed the apparent convention to put environment id in a header. But I'm not very experienced with REST conventions so I have no objections to applying your suggestion. Could you confirm you think that's best?

Reply:
It looks like a good idea to me. This put more structure into the API. We do the same thing with agents for example: `/agent/<name>/<action>`. @bartv Do you agree?

Reply:
I agree partially. Since about a year nested api paths are allowed so we should indeed introduce some more structure. 

At some point we / I decided to put the environment (TID == Tenant ID) in a header and not in the path. The reasoning between headers vs path is:

- In header when you are operating in the context of an environment. You can also authorize per environment
- In path when there is no current env and you are operating on an environment.

 This has advantages and disadvantages but I would like to stay consistent with most others. In this case it means that we should not put the env in the url. I do like to have more structure, but creating `/environment/halt` next to `/environment/<id>` is a bit too complicated.

Do anyone have suggestions?

Reply:
The middle ground between consistency and informative might be to go for `/environment_halt` or `/halt_environment`?

Reply:
That really looks like a Belgian compromise ;-)

Reply:
I'm afraid it's all I've got at the moment.

Reply:
/actions/halt
/actions/resume

Reply:
Not `/actions/environment/halt`?

Reply:
That is actually even better.